mdtraj/geometry/sasa.py

```diff
@@ -184,7 +203,10 @@ def shrake_rupley(traj, probe_radius=0.14, n_sphere_points=960, mode='atom', cha
         atom_radii = [_ATOMIC_RADII[atom.element.symbol] for atom in traj.topology.atoms]
     radii = np.array(atom_radii, np.float32) + probe_radius
 
-    _geometry._sasa(xyz, radii, int(n_sphere_points), atom_mapping, out)
+    _geometry._sasa(xyz, radii, int(n_sphere_points), atom_mapping, atom_indices, out)
+
+    # Cast np.nan for residues/atoms not covered by atom_indices
+    out[:, ~np.isin(np.unique(atom_mapping[np.arange(traj.n_atoms)]), np.unique(atom_mapping[atom_indices]))] = np.nan
```

Comment:
The tilda here is just to flip booleans?

Reply:
Yep. If you prefer I can use np.invert. In any case, from the [docs](https://numpy.org/doc/stable/reference/generated/numpy.invert.html)

> The ~ operator can be used as a shorthand for np.invert on ndarrays.